src/middlewares/has-permission.js

```diff
@@ -0,0 +1,27 @@
+const User = require("../models/user");
+const Permission = require("../models/permission")
+
+module.exports = (permission) => {
```

Comment:
Nessa função deveria liberar tbm se for o owner, independente de ter a permissão ou não